src/main/java/world/bentobox/parkour/listeners/CourseRunnerListener.java

```diff
@@ -71,11 +72,40 @@ public void onVisitorArrive(IslandEnterEvent e) {
         } else if (!parkourRunManager.timers().containsKey(e.getPlayerUUID())) {
             user.notify("parkour.to-start");
         }
-        if (island.getFlag(addon.PARKOUR_CREATIVE) <= island.getRank(user)) {
-            user.setGameMode(GameMode.CREATIVE);
-        } else {
-            user.setGameMode(GameMode.SURVIVAL);
+
+        if (addon.inWorld(user.getLocation())) {
```

Comment:
This includes nether and end. So if you are trying to keep it to the same world then you'll need to check some other things too.

Reply:
It probably doesn't matter here, since both the to and the from world's are managed by parkour

Reply:
clarified comments a little